exporters/elasticsearch/src/es_log_exporter.cc

```diff
@@ -0,0 +1,202 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/exporters/elasticsearch/es_log_exporter.h"
+#include "opentelemetry/exporters/elasticsearch/es_log_recordable.h"
+
+namespace nostd       = opentelemetry::nostd;
+namespace sdklogs     = opentelemetry::sdk::logs;
+namespace http_client = opentelemetry::ext::http::client;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+/**
+ * This class handles the response message from the Elasticsearch request
+ */
+class ResponseHandler : public http_client::EventHandler
+{
+public:
+  /**
+   * Automatically called when the response is received, store the body into a string and notify any
+   * threads blocked on this result
+   */
+  void OnResponse(http_client::Response &response) noexcept override
+  {
+    // Store the body of the request
+    body_ = std::string(response.GetBody().begin(), response.GetBody().end());
+
+    // Set the response_received_ flag to true and notify any threads waiting on this result
+    response_received_ = true;
+    cv_.notify_all();
+  }
+
+  /**
+   * A method the user calls to block their thread until the response is received, or the timeout is
+   * exceeded.
+   */
+  bool waitForResponse(unsigned int timeoutSec = 1)
+  {
+    std::mutex mutex_;
+    std::unique_lock<std::mutex> lk(mutex_);
+    cv_.wait_for(lk, std::chrono::milliseconds(1000 * timeoutSec));
+    return response_received_;
+  }
+
+  /**
+   * Returns the body of the response
+   */
+  std::string GetResponseBody()
+  {
+    if (!response_received_)
+      return "No response";
+
+    return body_;
+  }
+
+  // Virtual method definition that isn't used
+  void OnEvent(http_client::SessionState state,
+               opentelemetry::nostd::string_view reason) noexcept override
+  {}
+
+private:
+  // Define a condition variable used for blocking
+  std::condition_variable cv_;
+
+  // Whether the response from Elasticsearch has been received
+  bool response_received_ = false;
+
+  // A string to store the response body
+  std::string body_ = "";
+};
+
+ElasticsearchLogExporter::ElasticsearchLogExporter()
+    : options_{ElasticsearchExporterOptions()},
+      session_manager_{new ext::http::client::curl::SessionManager()}
+{}
+
+ElasticsearchLogExporter::ElasticsearchLogExporter(const ElasticsearchExporterOptions &options)
+    : options_{options}, session_manager_{new ext::http::client::curl::SessionManager()}
+{}
+
+std::unique_ptr<sdklogs::Recordable> ElasticsearchLogExporter::MakeRecordable() noexcept
+{
```

Comment:
nit: if the exporter is shut down, does it still need to make a recordable? ie. could it return `nullptr`?

Reply:
Saw your discussion on ostream, leaving as is